drivers/sensor/renesas/hs400x/Kconfig

```
@@ -0,0 +1,11 @@
+# HS400X temperature and humidity sensor
+
+# Copyright (c) 2024 Renesas Electronics Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config HS400X
+	bool "HS400x sensor"
+	default y
+	depends on DT_HAS_RENESAS_HS400X_ENABLED
```

```suggestion
 depends on DT_HAS_RENESAS_HS400X_ENABLED
 select CRC
```

There are many sensors that make CRC check optional and something that can be enabled/disabled via ab additional Kconfigs option. I'll let sensor maintainers comment on whether that's preferable to forcing users to pull CRC code if they don't want or can't afford it (it feels to me like it would indeed be better, and is likely a reasonably trivial change)

---

It also seems more preferable to me to allow the user to decide, thank you for the suggestion. I've added a Kconfig option for this.